libraries/mbed/targets/hal/TARGET_NORDIC/TARGET_MCU_NRF51822/sleep.c

```diff
@@ -28,5 +28,5 @@ void sleep(void)
 void deepsleep(void)
 {
     sleep();
-    //   NRF_POWER->SYSTEMOFF=1;
+    NRF_POWER->SYSTEMOFF = 1;
```

Comment:
This is not good if an app is using softdevice. I believe it should be sd_power_system_off() ? Did you test this change?